mongoengine/queryset/utils.py

```diff
@@ -0,0 +1,23 @@
+from functools import wraps
+
+from pymongo.errors import AutoReconnect
+
+
+def retry(func, tries=5, exceptions=(AutoReconnect,)):
```

Comment:
I'd prefer if this was less ambiguous. In the spirit of starting with concrete and specific code, and abstracting it away only when needed, how about we call it `retry_upon_reconnect` and hard-code the exception? If it was a generic `retry` wrapper 1) it wouldn't be in `mongoengine/queryset/utils.py`, 2) the exception list should be a parameter (so that I can do `@retry(on=AutoReconnect)`).